airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,550 @@
+        name: str,
+        flow_update: bool,
+        source_field: Optional[str] = None,
+        filter_date: Optional[str] = None,
+        poll_interval: int = 20,
+        aws_conn_id: str = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        if source not in SUPPORTED_SOURCES:
+            raise ValueError(f"{source} is not a supported source (options: {SUPPORTED_SOURCES})!")
+        self.filter_date = filter_date
+        self.name = name
+        self.source = source
+        self.source_field = source_field
+        self.poll_interval = poll_interval
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+        self.flow_update = flow_update
+
+    @cached_property
+    def hook(self) -> AppflowHook:
+        """Create and return an AppflowHook."""
+        return AppflowHook(aws_conn_id=self.aws_conn_id, region_name=self.region)
+
+    def execute(self, context: "Context") -> None:
+        self.filter_date_parsed: Optional[datetime] = (
+            datetime.fromisoformat(self.filter_date) if self.filter_date else None
+        )
+        if self.flow_update:
+            self._update_flow()
+        self._run_flow(context)
+
+    def _get_connector_type(self, response: "DescribeFlowResponseTypeDef") -> str:
+        connector_type = response["sourceFlowConfig"]["connectorType"]
+        if self.source != connector_type.lower():
+            raise ValueError(f"Incompatible source ({self.source} and connector type ({connector_type})!")
+        return connector_type
+
+    def _update_flow(self) -> None:
+        response = self.hook.conn.describe_flow(flowName=self.name)
+        connector_type = self._get_connector_type(response)
+
+        # cleanup
+        tasks: List["TaskTypeDef"] = []
+        for task in response["tasks"]:
+            if (
+                task["taskType"] == "Filter"
+                and task.get("connectorOperator", {}).get(connector_type) != "PROJECTION"
+            ):
+                self.log.info("Removing task: %s", task)
+            else:
+                tasks.append(task)  # List of non-filter tasks
+
+        self._add_filter(connector_type, tasks)
+
+        # Clean up to force on-demand trigger
+        trigger_config = copy.deepcopy(response["triggerConfig"])
+        del trigger_config["triggerProperties"]
+
+        self.hook.conn.update_flow(
+            flowName=response["flowName"],
+            destinationFlowConfigList=response["destinationFlowConfigList"],
+            sourceFlowConfig=response["sourceFlowConfig"],
+            triggerConfig=trigger_config,
+            description=response.get("description", "Flow description."),
+            tasks=tasks,
+        )
+
+    def _add_filter(self, connector_type: str, tasks: List["TaskTypeDef"]) -> None:  # Interface
+        pass
+
+    def _run_flow(self, context) -> str:
+        ts_before: datetime = datetime.now(timezone.utc)
```

Comment:
I can't use `time.monotonic()` in that case because it is not an elapsed time measure situation. I really need to get the UTC time to compare to the UTC time returned further by Amazon AppFlow.

Is it OK?

Reply:
Ah I see, makes sense.